protocol.md

```diff
@@ -1836,6 +1845,27 @@ interface CompletionItem {
 	 */
 	insertTextFormat?: InsertTextFormat;
 	/**
+	 * Whether automatic relative indentation fixes are applied to a multi-line insert text.
+	 * This option applies to both the `insertText` property and the `newText` property
+	 * of a provided `textEdit`.
+	 *
+	 * Relative indentation fixes are applied as follows:
+	 *
+	 * If the insert text contains no line breaks, no fixes are applied.
+	 *
+	 * Otherwise, define the reference line as the line in document where the insert text
+	 * is being applied.
+	 *
+	 * The reference indent is determined by taking all the whitespace
+	 * characters from the front of the reference line upto either the first non-whitespace
+	 * character; or the start position where the insert text is being applied. Whichever
+	 * of these two yields the shortest string is taken as the reference indent.
+	 *
+	 * To adjust indentation, the reference indent is then inserted into the insert text right
+	 * after every line break.
+	 */
+	relativeIndent?: boolean
```

Comment:
I don't feel strongly about it either way. However defaulting to not supporting magic indentations is probably what most people would assume if they read the current LSP description as it make no mention of these magic indents at all. Therefore, any implementor working from reading the spec would have no clue they are supposed to support these magic indents.

So in a way, I'd argue that vscode, in this case is actually not complying with the spec, as it is currently written.